WireCells/Tests/WireCellsTests/WireCellsTests.swift

```diff
@@ -0,0 +1,93 @@
+        static let rootPath = URL(string: "wire-cells-ios")!
+        // Replace `nil` with the secret key, which can be obtained on 1Password ("Wire Cells demo secret key")
+        static let secretKey: String! = nil
+    }
+
+    var wireCellsService: WireCellsService!
+
+    override func setUp() async throws {
+        guard Constants.secretKey != nil else {
+            fatalError("Please provide a secret key")
+        }
+        wireCellsService = try WireCellsServiceImplementation(
+            endpointURL: Constants.endpointURL,
+            rootPath: Constants.rootPath,
+            secretKey: Constants.secretKey
+        )
+    }
+
+    override func tearDown() {
+        wireCellsService = nil
+    }
+
+    func testUploadFile() async throws {
+        let data = "Hello World!".data(using: .utf8)!
+        for await result in wireCellsService.uploadFiles([
+            WireCellsFileUploadInfo(data: data, uploadPath: "HelloWorld.txt")
+        ]) {
+            switch result {
+            case .started:
+                break
+            case .uploading:
+                break
+            case let .success(fileUploadInfo, uploadedNode):
+                XCTAssert(uploadedNode.path.absoluteString == "\(Constants.rootPath)/\(fileUploadInfo.uploadPath)")
+            case let .failure(_, error):
+                XCTFail(error.localizedDescription)
+            }
+        }
+    }
+
+    func testListFilesRoot() async throws {
+        // XCTest Documentation
+        // https://developer.apple.com/documentation/xctest
+
+        // Defining Test Cases and Test Methods
+        // https://developer.apple.com/documentation/xctest/defining_test_cases_and_test_methods
+
+        do {
+            let results = try await wireCellsService.listFiles()
+            XCTAssert(results.count == 3)
+        } catch {
+            XCTFail(error.localizedDescription)
+        }
+    }
+
+    func testListFilesEmptyFolder() async throws {
+        // XCTest Documentation
+        // https://developer.apple.com/documentation/xctest
+
+        // Defining Test Cases and Test Methods
+        // https://developer.apple.com/documentation/xctest/defining_test_cases_and_test_methods
+
+        do {
+            let results = try await wireCellsService.listFiles(atPath: "EmptyFolderTree/EmptyFolderTreeLeaf")
+            XCTAssert(results.isEmpty)
+        } catch {
+            XCTFail(error.localizedDescription)
+        }
+    }
+
+    func testListFilesEmptyFolderTree() async throws {
+        // XCTest Documentation
+        // https://developer.apple.com/documentation/xctest
+
+        // Defining Test Cases and Test Methods
+        // https://developer.apple.com/documentation/xctest/defining_test_cases_and_test_methods
+
+        do {
+            let results = try await wireCellsService.listFiles(atPath: "EmptyFolderTree")
+            XCTAssert(results.count == 1)
+        } catch {
+            XCTFail(error.localizedDescription)
+        }
```

Comment:
As test method throws, no need for the do/ catch if it fails the test will fail
```suggestion
            let results = try await wireCellsService.listFiles(atPath: "EmptyFolderTree")
            XCTAssert(results.count == 1)
```

Reply:
Yes. But I want to pick what is printed when it fails.